packages/errors/index.ts

```diff
@@ -0,0 +1,37 @@
+class MongoshRuntimeError extends Error {
+  constructor(...args) {
+    super(...args);
+    this.name = 'MongoshRuntimeError';
+  }
+}
+
+class MongoshInternalError extends Error {
+  constructor(msg, opts?) {
+    super(opts);
```

Comment:
`super` is the usual constructor of error, which takes the message. Should be like this:

```
constructor(msg) {
   super(
     msg + 
    '\nThis is an error inside Mongosh. Please file a bug report. ' + 
     'Please include a log file from this session.');
   this.name = 'MongoshInternalError'
}
```

Reply:
from the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error), it seems to super the second argument.

<img width="1111" alt="Capture d’écran, le 2020-04-24 à 14 09 14" src="https://user-images.githubusercontent.com/8107784/80211086-4a876f80-8635-11ea-942f-5ea630d9d06a.png">

Reply:
I think that example is confusing, but proves that super takes a message as first parameter, just in a super-weird way. If you check better: the first parameter of the CustomError constructor (foo) is not the message and is not passed to `super`. The rest of arguments is.

So when is used we get:

```
 new CustomError(
  'baz',  // this gets assigned to e.foo
  'bazMessage' // this is going to be the first argument of `...params`, 
                             // so the first argument of super() and it populates 
                             //  e.message` since is the message.  
```

An easier example would be this one:

```
class PoliteError extends Error {
  constructor(reason) {
    super(`We are really sorry but something went wrong: ${reason}.`)
  }
}

try {
  throw new PoliteError('life is hard');
} catch (e) {
  console.log(e.message);
}
  
// => We are really sorry but something went wrong: life is hard.
```